lib/api/1.1/northbound/router.js

```diff
@@ -0,0 +1,61 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+var di = require('di'),
+    express = require('express');
+
+module.exports = northApiFactory;
+
+di.annotate(northApiFactory, new di.Provide('northbound-api-router'));
+di.annotate(northApiFactory, new di.Inject(
+        di.Injector
+    )
+);
+
+function northApiFactory (injector) {
+    var router = express.Router();
+    router.endpoint = {};
+
+    router.mount = function() {
+
+        // Un-mount existing routes in case router.mount has
+        // been call multiple times.
+        this.unMountAllRouters();
```

Comment:
Thanks Brian for the comments. The intention here is to unmount everything that has already been mounted, when mount is called. The user case is that in the unit test, we would want to change a configure file (ie, authEnabled = True change to authEnabled = false) and then restart the http-service. The routers(ie, /login) that mounted when authEnable=True should be unmounted after restarting the http-service. 

I would take the action to update the notes to reflect my intention here.